### PR TITLE
fix(docs): Update persisting-store-data.md

### PR DESCRIPTION
**Pull request by charkour**
Add context for TypeScript users. Related to https://github.com/pmndrs/zustand/issues/1127#issuecomment-1193922792

### CODE REVIEW

File: docs/persisting-store-data.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ export const useFishStore = create(
 )
 ```
 
+⚠️ This readme is written for JavaScript users. Basic TypeScript usage doesn't require anything special except for writing `create<State>()(...)` instead of `create(...)`. A more complete TypeScript guide is [here](https://github.com/pmndrs/zustand/blob/main/docs/typescript.md).
```
Comment (dai-shi) [inline docs/persisting-store-data.md:28]: Hm, this "doc" is written somewhat in TypeScript. So, it doesn't sound correct.

Comment (charkour) [reply docs/persisting-store-data.md:28]: Sounds good, I'll update it.

**Comment by dai-shi** [comment]
Let's wait for @AnatoleLucet review.

**Comment by charkour** [comment]
Sounds good. Thank you!

I can make any requested changes.

**Review by AnatoleLucet — APPROVED** [review]
Looks good! Thanks

**Comment by devanshj** [comment]
IMO [this older version](https://github.com/pmndrs/zustand/pull/1136/commits/a21293060bf7f4fcf99d03f81144eec2f7ac3589) was better, it's important to link/redirect to the typescript guide instead of copying/applying it here (so that if at all when typescript usage changes we'll have to change it at only one place, plus the guide is more complete).

> Hm, this "doc" is written somewhat in TypeScript. So, it doesn't sound correct.

How about the following?

⚠️ Most code snippets in this doc are written in JavaScript. Refer [the TypeScript guide](TypeScript ) for usage in TypeScript.

Just my two cents, the current version is not bad too.

**Comment by dai-shi** [comment]
Let's wait for more feedback from users. I think what people want is more dedicate TS guide specific for the persist middleware.

**Pull request merged**